jetty-websocket/websocket-client/src/main/java/org/eclipse/jetty/websocket/client/WebSocketClient.java

```diff
@@ -378,21 +381,37 @@ public WebSocketClient(final WebSocketContainerScope scope, EventDriverFactory e
         if (LOG.isDebugEnabled())
             LOG.debug("connect websocket {} to {}",websocket,toUri);
 
-        init();
-
```

Comment:
Delayed init must still occur.
This is important in large webapp deployments (think 20+ webapps).
Its likely that not all webapps deployed in a JSR356 environment will even use the client features.

Reply:
The init is pretty much a no-op in this case and it's a private method. All it does is register itself with the shutdown thread, which is adding a reference to itself to CopyOnWriteArrayList. I can refactor this if it's a performance concern.

In a large webapp deployment is there something that instantiates WebSocketClients if they're not being used?

Reply:
Yes.  Unfortunately, the way that JSR356 (aka javax.websocket) is specified, a ServerContainer extends a ClientContainer.

See: 
 * http://docs.oracle.com/javaee/7/api/javax/websocket/server/ServerContainer.html
 * http://docs.oracle.com/javaee/7/api/javax/websocket/WebSocketContainer.html

This means that every JSR356 ServerContainer is also a client and needs a WebSocketClient, but we attempt to delay the start of the WebSocketClient (and associated HttpClient) until such time that an actual client `connectToServer()` is attempted.

That init() method is one of the ways that the delay is pulled off.